webpack.config.js

```diff
@@ -71,7 +60,13 @@ const config = {
 		json3: "JSON", // socket.io uses json3.js, but we do not target any browsers that need it
 	},
 	plugins: [
-		new webpack.optimize.CommonsChunkPlugin("js/bundle.vendor.js")
+		// socket.io uses debug, we don't need it
+		new webpack.NormalModuleReplacementPlugin(/debug/, path.resolve(__dirname, "scripts/noop.js")),
+		// automatically split all vendor dependancies into a separate bundle
+		new webpack.optimize.CommonsChunkPlugin({
+			name: "js/bundle.vendor.js",
+			minChunks: (module) => module.context && module.context.indexOf("node_modules") !== -1
```

Comment:
It's based on requires. Basically it generates identical bundles to what master has now.